File: src/com/frickerg/demo/notification/NotificationTest.java
package com.frickerg.demo.notification;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Date;

import static org.junit.Assert.assertEquals;

public class NotificationTest {

	private Notification notification;

	@Before
	public void setUp() throws Exception {
		this.notification = new Notification("message",new Date());
	}

	@After
	public void tearDown() throws Exception {
	}

	@Test
	public void getMessage() {
		assertEquals("message", this.notification.getMessage());
	}

	@Test
	public void setMessage() {
		String changeString = "new_message";
		this.notification.setMessage(changeString);

		System.out.println(this.notification.getMessage());
		assertEquals(changeString, this.notification.getMessage());
	}

	@Test
	public void getTimestamp() {
	}

	@Test
	public void setTimestamp() {
	}

	@Test
	public void testToString() {
	}

	@Test
	public void testEquals() {
	}

	@Test
	public void testHashCode() {
	}
}

File: src/com/frickerg/demo/notification/NotificationTest.kt
package com.frickerg.demo.notification

import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import java.util.*

class NotificationTest {
    private var notification: Notification? = null
    @Before
    @Throws(Exception::class)
    fun setUp() {
        notification = Notification("message", Date())
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @get:Test
    val message: Unit
        get() {
            Assert.assertEquals("message", notification!!.message)
        }

    @Test
    fun setMessage() {
        val changeString = "new_message"
        notification!!.message = changeString
        println(notification!!.message)
        Assert.assertEquals(changeString, notification!!.message)
    }

    @get:Test
    val timestamp: Unit
        get() {}

    @Test
    fun setTimestamp() {
    }

    @Test
    fun testToString() {
    }

    @Test
    fun testEquals() {
    }

    @Test
    fun testHashCode() {
    }
}